src/main/kotlin/io/bazel/worker/JavaPersistentWorker.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package io.bazel.worker
+
+import com.google.devtools.build.lib.worker.WorkerProtocol.WorkRequest
+import com.google.devtools.build.lib.worker.WorkerProtocol.WorkResponse
+import src.main.kotlin.io.bazel.worker.GcScheduler
+import java.io.InputStream
+import java.time.Duration
+import java.util.concurrent.ExecutorCompletionService
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicLong
+
+/**
+ * PersistentWorker satisfies Bazel persistent worker protocol for executing work.
+ *
+ * Supports multiplex (https://docs.bazel.build/versions/master/multiplex-worker.html) provided
+ * the work is thread/coroutine safe.
+ *
+ * @param executor thread pool for executing tasks.
+ * @param captureIO to avoid writing stdout and stderr while executing.
+ * @param cpuTimeBasedGcScheduler to trigger gc cleanup.
+ */
+class JavaPersistentWorker(
```

Comment:
Why the rename here? Is this naming relevant to "java vs. js or llvm/native"? Or java in some other sense?

Reply:
Ah, I had named the other Kotlin for coroutines. Will fix.